timeutil/schedule.go

```diff
@@ -39,6 +39,30 @@ func (t TimeOfDay) String() string {
 	return fmt.Sprintf("%02d:%02d", t.Hour, t.Minute)
 }
 
+// Sub subtracts `other` TimeOfDay from current and returns duration
+func (t TimeOfDay) Sub(other TimeOfDay) time.Duration {
+	t1 := time.Duration(t.Hour)*time.Hour + time.Duration(t.Minute)*time.Minute
+	t2 := time.Duration(other.Hour)*time.Hour + time.Duration(other.Minute)*time.Minute
+	return t1 - t2
+}
+
+// Add adds given duration and returns a new TimeOfDay
+func (t TimeOfDay) Add(dur time.Duration) TimeOfDay {
+	t1 := time.Duration(t.Hour)*time.Hour + time.Duration(t.Minute)*time.Minute
+	t2 := t1 + dur
+	nt := TimeOfDay{
+		Hour:   int(t2.Hours()) % 24,
```

Comment:
Is this accurate for stuff like leap year and 60ths second? I would prefer if we could work at a level of simple seconds and convert to/from other representations.

Reply:
Thanks for the commen. There's no date and location context so it's not possible for this to be aware of leap years, leap seconds (`time.Time()` is not aware of leap seconds either) or DST. It's using time.Duration() under the hood (which is just seconds), the calculations are done using `time.Hour` and `time.Minute` both are convertible to integer types without loss. 

IMO this should be fine for the purpose of just adding some duration to `TimeOfDay` and wrapping around.